external/CMakeLists.txt

```diff
@@ -54,6 +54,16 @@ ExternalProject_Add(noise-c
 add_library(noiseprotocol STATIC IMPORTED GLOBAL)
 set_property(TARGET noiseprotocol
   PROPERTY IMPORTED_LOCATION ${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}/libnoiseprotocol.a)
+set_target_properties(noiseprotocol
+  PROPERTIES
+    INTERFACE_INCLUDE_DIRECTORIES
```

Comment:
TIL!

Though still not sure how it works, are those automatically added as include headers by add_dependency? 

https://cmake.org/cmake/help/v3.15/prop_tgt/INTERFACE_INCLUDE_DIRECTORIES.html is very cryptic in how this works.

Reply:
heh, it is even more weird they are added as `include_directories` because of `target_link_libraries`. `add_dependency` only forces those targets to be built before this target is built. `add_dependency` is a hack becuase the dependencies aren't CMake projects.

You can read more about it here: https://cmake.org/cmake/help/v3.10/command/target_include_directories.html